internal/testutil/test_setup.go

```diff
@@ -144,3 +150,34 @@ func OnlyEnableCode() {
 	config.CurrentConfig().SetSnykOssEnabled(false)
 	config.CurrentConfig().SetSnykCodeEnabled(true)
 }
+
+func SetupCustomTestRepo(t *testing.T, rootDir string, url string, targetCommit string, logger *zerolog.Logger) (string, error) {
+	t.Helper()
+	tempDir := filepath.Join(rootDir, util.Murmur(t.Name()))
+	assert.NoError(t, os.MkdirAll(tempDir, 0755))
+	repoDir := "1"
+	absoluteCloneRepoDir := filepath.Join(tempDir, repoDir)
+	cmd := []string{"clone", url, repoDir}
+	logger.Debug().Interface("cmd", cmd).Msg("clone command")
+	clone := exec.Command("git", cmd...)
```

Comment:
should we use the repo setup logic from go-git here? 
Sample: git_cloner.go -> initGitRepo

Reply:
Actually maybe it doesn't make much sense for this case. We can revisit it later

Reply:
Yeah, we can consolidate that.